src/gpu/intel/ref_matmul.cpp

```diff
@@ -221,32 +221,38 @@ status_t ref_matmul_t::execute_ref(const exec_ctx_t &ctx) const {
     arg_list.set(arg_idx++, b);
     arg_list.set(arg_idx++, subbyte_pack ? *tmp : c);
     arg_list.set(arg_idx++, bias);
-    arg_list.set(arg_idx++, a0);
+    arg_list.set_mem_or_host_scalar<int>(arg_idx++, a0,
+            pd()->attr()->zero_points_.get(DNNL_ARG_SRC).is_host_scalar());
```

Comment:
Can we hide the memory/host logic inside of `arg_list.set()`?

It would be nice to have something like:

```c++
// host_scalar_type has to be a function parameter vs a template parameter as in general
// we can have scalars of non-fixed types.
void set(int index, const memory_storage_t &storage, data_type_t host_scalar_type = data_type::undef) {
    if (!storage.is_host_scalar()) {
        args_[index].set_value(storage);
        return;
    }
    auto *host_scalar = static_cast<...>(&storage);
    switch (host_scalar_type) {
        case data_type::s32: {
            int32_t value = 0;
            host_scalar->get_scalar_value(&value, sizeof(value));
            set(index, value);
            break;
        }
        // ...
    }
}
```

Reply:
I'm not attached to having an api with different name, so I will update as you suggested.

Sure, I'll take a look into switching from size to dt to make it more convenient. I don't see any immediate issue with that.

Reply:
I've refactored/merged setting from storage, please take a look